Samples/Islands/cs-winui-packaged/WinUIDesktopAppWithIsland/MainWindow.xaml.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.UI.Content;
+using Microsoft.UI.Composition;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Navigation;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Microsoft.UI.Xaml.Hosting;
+using System.Numerics;
+
+// To learn more about WinUI, the WinUI project structure,
+// and more about our project templates, see: http://aka.ms/winui-project-info.
+
+namespace WinUIDesktopAppWithIsland
+{
+    /// <summary>
+    /// An empty window that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class MainWindow : Window
+    {
+        public MainWindow()
+        {
+            this.InitializeComponent();
+        }
+
+        List<object> _keepAlive = new List<object>();
+        int idx = 0;
+
+        Microsoft.UI.Xaml.Shapes.Rectangle GetNextHostElement()
+        {
+            if (idx < _rectanglePanel.Children.Count)
+            {
+                return ((Microsoft.UI.Xaml.Shapes.Rectangle)_rectanglePanel.Children[idx++]);
+            }
+
+            return null;
+        }
+
+        public async void SetupHelmet()
+        {
+            ContentIsland parentIsland = this.Content.XamlRoot.TryGetContentIsland();
```

Comment:
I'll clean this up in a subsequent PR to create a proper ContentHost control that can integrate with layout.

Reply:
OK, sounds good.